### PR TITLE
Add HTTP server restart params into node config [ECR-3748]

### PR DESCRIPTION
**Pull request by SergeiMal**
## Overview

---

See: https://jira.bf.local/browse/ECR-3748

### Definition of Done

- [ ] There are no TODOs left in the merged code
- [ ] Change is covered by automated tests
- [ ] Benchmark results are attached (if applicable)
- [ ] The [coding guidelines] are followed
- [ ] Public API has proper documentation
- [ ] Changelog is updated if needed (in case of notable or breaking changes)
- [ ] The continuous integration build passes

[coding guidelines]: https://github.com/exonum/exonum/blob/master/CONTRIBUTING.md#conventions

### CODE REVIEW

File: exonum/src/api/manager.rs [hunk 1]
```diff
@@ -136,12 +132,12 @@ impl Handler<StartServer> for ApiManager {
             Ok(addr) => addr,
             Err(e) => {
                 warn!("Error handling service start {:?}: {}", msg.config, e);
-                if msg.attempt == Self::RETRY_ATTEMPTS {
+                if msg.attempt == self.runtime_config.service_retry_attempt as usize {
```
Comment (popzxc) [inline exonum/src/api/manager.rs]: Maybe make `attempt` `u64`  as well to remove this cast?

File: exonum/src/events/network.rs [hunk 2]
```diff
@@ -76,6 +76,21 @@ pub enum NetworkRequest {
     Shutdown,
 }
 
+#[derive(Serialize, Deserialize, Debug, Clone, Copy, PartialEq)]
+pub struct HttpBackendConfig {
+    pub restart_retry_attempts: u64,
```
Comment (popzxc) [inline exonum/src/events/network.rs]: Hm. IMO `u16` is enough in both cases.

Comment (aleksuss) [reply exonum/src/events/network.rs]: It would be good to add a comment doc for a public field.

File: test-suite/testkit/src/lib.rs [hunk 3]
```diff
@@ -690,6 +690,8 @@ impl TestKit {
                 ApiRuntimeConfig::new(private_api_address, ApiAccess::Private),
             ],
             api_aggregator,
+            service_retry_attempt: 20,
```
Comment (popzxc) [inline test-suite/testkit/src/lib.rs]: Hm... Very minor nit: not sure if we should have that many attempts in the testkit. I'd suggest 5 or even 3.

@aleksuss @alekseysidorov what's your opinion on this?

File: CHANGELOG.md [hunk 4]
```diff
@@ -65,6 +65,12 @@ The project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html)
 
 ### New Features
 
+#### exonum
+
+- New config params `http_backend_config.restart_retry_attempts` and
+  `http_backend_config.restart_retry_interval` added into `NetworkConfiguration`.
+  They are intend to configure of the HTTP-server restart settings (#1536).
```
Comment (aleksuss) [inline CHANGELOG.md]:
```suggestion
  They are intended to configure of the HTTP-server restart settings (#1536).
```

File: test-suite/testkit/src/lib.rs (hunk 3 above)
Comment (aleksuss) [reply test-suite/testkit/src/lib.rs]: IMO it's not principle. 5 is ok.

File: exonum/src/events/network.rs [hunk 5]
```diff
@@ -96,6 +116,7 @@ impl Default for NetworkConfiguration {
             tcp_nodelay: true,
             tcp_connect_retry_timeout: 15_000,
```
Comment (slowli) [inline exonum/src/events/network.rs]: It may make sense to use uniform terminology ("interval" vs "timeout" and "max retries" vs "retry attempts").

**Pull request merged**